src/tools/r2rdump/R2RDump.cs

```diff
@@ -4,55 +4,436 @@
 
 using System;
 using System.Collections.Generic;
+using System.CommandLine;
+using System.IO;
 
 namespace R2RDump
 {
     class R2RDump
     {
-        public static void OutputWarning(string warning)
+        private bool _help = false;
+        private IReadOnlyList<string> _inputFilenames = Array.Empty<string>();
+        private string _outputFilename = null;
+        private bool _raw = false;
+        private bool _header = false;
+        private bool _disasm = false;
+        private IReadOnlyList<string> _queries = Array.Empty<string>();
+        private IReadOnlyList<string> _keywords = Array.Empty<string>();
+        private IReadOnlyList<int> _runtimeFunctions = Array.Empty<int>();
+        private IReadOnlyList<string> _sections = Array.Empty<string>();
+        private bool _diff = false;
+
+        private R2RDump()
+        {
+        }
+
+        private ArgumentSyntax ParseCommandLine(string[] args)
+        {
+            ArgumentSyntax argSyntax = ArgumentSyntax.Parse(args, syntax =>
+            {
+                syntax.ApplicationName = "R2RDump";
+                syntax.HandleHelp = false;
+                syntax.HandleErrors = false;
```

Comment:
Is there a reason HandleErrors if false? I'd expect it to be true

Reply:
if HandleErrors is true, when there's an invalid arg, the program gets Exception Unhandled instead of going into the catch block in the main function. Should I handle the exception there instead?

Reply:
What information is given to the user in that case? Is it clear to the user what went wrong?

Reply:
It does tell the user which argument is wrong, but there's a System.ExecutionEngineException if I don't handle it there